node/core/dispute-coordinator/src/real/initialized.rs

```diff
@@ -533,31 +588,45 @@ impl Initialized {
 				}
 			},
 			DisputeCoordinatorMessage::RecentDisputes(tx) => {
-				let recent_disputes = overlay_db.load_recent_disputes()?.unwrap_or_default();
-				let _ = tx.send(recent_disputes.keys().cloned().collect());
+				// Return error if session information is missing.
+				if let Some(subsystem_error) = self.error.clone() {
+					return Err(Error::NonFatal(NonFatal::RollingSessionWindow(subsystem_error)))
+				} else {
+					let recent_disputes = overlay_db.load_recent_disputes()?.unwrap_or_default();
+					let _ = tx.send(recent_disputes.keys().cloned().collect());
+				}
```

Comment:
If this is such a common pattern make it a `fn foo() -> Result<(),Error>` and call `foo()?`, readability is not very good.

Reply:
Yeah, that's what I was mentioning my own comment, this messes up readability completely.